src/Core/OrganizationFeatures/OrganizationGroups/DeleteGroupCommand.cs

```diff
@@ -0,0 +1,36 @@
+﻿using Bit.Core.Entities;
+using Bit.Core.Repositories;
+using Bit.Core.Services;
+
+namespace Bit.Core.OrganizationFeatures.OrganizationGroups;
+
+public class DeleteGroupCommand : IDeleteGroupCommand
+{
+    private readonly IGroupRepository _groupRepository;
+    private readonly IEventService _eventService;
+
+    public DeleteGroupCommand(IGroupRepository groupRepository, IEventService eventService)
+    {
+        _groupRepository = groupRepository;
+        _eventService = eventService;
+    }
+
+    public async Task DeleteAsync(Group group)
+    {
+        await _groupRepository.DeleteAsync(group);
+        await _eventService.LogGroupEventAsync(group, Enums.EventType.Group_Deleted);
+    }
+
+    public async Task DeleteManyAsync(IEnumerable<Group> groups)
+    {
+        var groupsToDelete = groups as Group[] ?? groups.ToArray();
```

Comment:
We'd probably be fine not converting to an array, but an `IEnumerable` does not guarantee it supports [multiple enumeration](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1851), so the conversion to an Array guarantees the `IEnumerable` is only enumerated a single time in the method. 

We could also change the parameter type to something like `IList` or `ICollection` if we wanted to avoid the need for a conversion, but it seemed like we typically pass `IEnumerable`s around.

Reply:
That makes sense, thanks! It may not be necessary once you make a bulk `LogGroupEventsAsync` method though, you'll only be enumerating it once in this method for the `Select` method.

Reply:
Unfortunately, it's still "required" as all methods (e.g. `LogGroupEventsAsync()`) that accept `IEnumerable` are, [by default](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1851#configure-the-default-assumption-of-methods-take-ienumerable-parameters), assumed to enumerate the argument.

But it is much cleaner now without the `for` loop and the multiple `awaits`